### PR TITLE
Implement transition-options to allow adding transitino-guard expressions

### LINKED ISSUE #85: Support workflow transition guard expressions
The guard-expressions should be set like this:

```
Transitions:
  publish (Private => Published) [guard-expr => here/transition_guard_publish]
  submit for publication (Private => Pending)
```

### PR DESCRIPTION
**Pull request by elioschmutz**
This RP adds a possibility to define transition-guard expressions.

closes #85

### CODE REVIEW

File: README.rst [hunk 1]
```diff
@@ -297,6 +297,30 @@ since our editors can edit, add new content, and request a review for existing
 content. Therefore, it is not necessary to distinguish Editor and Contributor
 role.
 
+Transition-Guard expressions
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Transition-guard expressions is a way to hide your transitions dynamically,
+in addition to the guard-roles. Use the options-syntax to define a guard-expression.
+
+Expressions in DCWorkflow are TALES expressions. To see the contextes provided
```
Comment (jone) [inline README.rst]: `contextes` => `contexts`
and maybe `provided for expressions` => `available in expressions`?

File: README.rst [hunk 2]
```diff
@@ -297,6 +297,30 @@ since our editors can edit, add new content, and request a review for existing
...
+for expressions, take a look at `portal_workflow/[your-workflow-id]/guardExprDocs`
+
+Warning: Transition-Guard do not protect the transition itself. If the user knows
```
Comment (jone) [inline README.rst]: `Transition-Guard` -> `Transition-guards`

File: ftw/lawgiver/wdl/parser.py [hunk 3]
```diff
@@ -257,3 +260,16 @@ def _get_consumers(self):
...
         return consumers
+
+    def _get_options(self, line):
+        options = {}
+        match = re.search(r'\[(.*?)\]', line)  # find everything between brackets
+        if not match:
+            return options
+
+        options_string = match.group(1)
+        for option_string in options_string.split('; '):
```
Comment (jone) [inline ftw/lawgiver/wdl/parser.py]: `'a=1; b=2'` is supported, but `a=1;b=2` not and `a=1 ; b=2` will probably not have the correct result.
I would change that to `.split(';')` and do a `.strip()` afterwards.

File: ftw/lawgiver/wdl/parser.py [hunk 4]
```diff
@@ -257,3 +260,16 @@ def _get_consumers(self):
...
+            option, value = option_string.split(' => ')
```
Comment (jone) [inline ftw/lawgiver/wdl/parser.py]: Same here: I'd handle spaces with `.strip()` and make it optional

File: README.rst [hunk 5]
```diff
@@ -297,6 +297,30 @@ since our editors can edit, add new content, and request a review for existing
...
+.. code:: rst
+
+    [My Custom Workflow]
+    Initial Status: Private
+
+    Status Private:
+    Status Published:
+
+    Transitions:
+      Publish (Private => Published) [guard-expr => python:here.guard(state_change)]
```
Comment (jone) [inline README.rst]: The code says `guard-expression`, but the docu says `guard-expr` 🤔

File: ftw/lawgiver/wdl/specification.py [hunk 6]
```diff
@@ -90,6 +90,7 @@ def __init__(self, title, src_status=None, dest_status=None,
         self._src_status_title = src_status_title or src_status.title
         self.dest_status = dest_status
         self._dest_status_title = dest_status_title or dest_status.title
+        self.options = options
```
Comment (jone) [inline ftw/lawgiver/wdl/specification.py]: The specification should also validate the input. In this case I'd expect it to detect when a make a typo (e.g. `guard_expr`) or when I use an unknown key (e.g. when I use the newest `ftw.lawgiver` docu but using an older version where this key does not yet exist).

I would do that with:
```python
unsupported_options = tuple(set(self.options) - {'guard-expression'})
if len(unsupported_options) == 0:
  raise ValueError('Unkown transition options: {!r}'.format(unsupported_options))
```

**Comment by elioschmutz** [comment]
@jone 👍 PR updated

**Comment by elioschmutz** [comment]
@jone could you take another look?

**Pull request merged**